File: cs/src/Ice/Acceptor.cs
// **********************************************************************
//
// Copyright (c) 2003-2009 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

namespace IceInternal
{

    using System;
    using System.Net.Sockets;

    public interface Acceptor
    {
        void close();
        void listen();
        IAsyncResult beginAccept(AsyncCallback callback, object state);
        Transceiver endAccept(IAsyncResult result);
        string ToString();
    }

}

File: cpp/src/Ice/Acceptor.cpp
// **********************************************************************
//
// Copyright (c) 2003-present ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#include <Ice/Acceptor.h>

using namespace std;
using namespace Ice;
using namespace IceInternal;

Acceptor::~Acceptor()
{
    // Out of line to avoid weak vtable
}

IceUtil::Shared* IceInternal::upCast(Acceptor* p) { return p; }
